src/components/NewItemForm.jsx

```diff
@@ -55,14 +101,110 @@ const NewItemForm = () => {
   const [endpoint, setEndpoint] = useState("");
   const [password, setPassword] = useState("");
   const [username, setUserName] = useState("");
+  const [itemNameError, setItemNameError] = useState(clearError);
+  const [endpointError, setEndpointError] = useState(clearError);
+  const [passwordError, setPasswordError] = useState(clearError);
+  const [usernameError, setUserNameError] = useState(clearError);
+
   const [collectionID, setCollectionID] = useState("");
+  const [collectionArray, setCollectionArray] = useState([]);
+  const [defaultCollectionsID, setDefaultCollectionID] = useState("");
 
   const handleCollectionChange = (event) => {
     setCollectionID(event.target.value);
   };
 
-  const onSaveVaultItem = () => {
-    console.log("status");
+  const getVaultCollections = async () => {
+    try {
+      let response = await getCollections(getCollections);
+      setCollectionArray(response.data);
+
+      for (let i = 0; i < response.data.length; i++) {
+        let collection = response.data[i];
+        if (collection.name == "Default") {
+          setDefaultCollectionID(collection.uuid);
+          setCollectionID(collection.uuid);
+        }
+      }
+    } catch (error) {
+      if (error.response.status === 403) {
+        alert("Please sign in again to continue");
+        navigate("/login-signup");
+        return;
+      }
+
+      if (error.response.status === 400) {
+        alert("Failed to access data, try again later");
+        window.location.reload();
+        return;
+      }
+    }
+  };
+
+  useEffect(() => {
+    getVaultCollections();
+  }, []);
+
+  const onSaveVaultItem = async () => {
+    let errorStatus = false;
+    errorStatus = isNullCheck(itemName, setItemNameError);
+    errorStatus = isNullCheck(password, setPasswordError);
+    errorStatus = isNullCheck(endpoint, setEndpointError);
+    errorStatus = isNullCheck(username, setUserNameError);
+
+    if (errorStatus) {
+      return;
+    }
+
+    let vaultItemObject = {
+      name: itemName,
+      url: endpoint,
+      username: username,
+      password: password,
+    };
+
+    console.log("Saving: ", vaultItemObject);
+    // try {
+    //   let vaultItemObject = {
+    //     name: itemName,
+    //     url: endpoint,
+    //     username: username,
+    //     password: password,
+    //   };
+    //   // figure out symmetricKey
+    //   let encryptVaultItem = await encryptVaultItem(
+    //     vaultItemObject,
+    //     "symmetricKey"
+    //   );
```

Comment:
The server will return a newly created vault item if the response is a success. In a previous conversation we were saying we should only update the UI when the response is a success. How were you thinking about handling updating the UI when creating new vault items and vault collections?

Reply:
One possible option I could think would work is to extract the POST request onto the parent and pass in the data as props. This way we can access both the new item and the collection in the dashboard (assuming that's where we keep it). I'm not sure if that's the best option, but probably the one that comes to mind. Open to suggestions!

Reply:
As we talked about in our FE sync today, we will pass the data down as props to the components that need the data. We will pessimistically update the UI. This will need to be done after the GET and UI Layout for displaying VC and VI are complete.